internal/bsr/testing.go

```diff
@@ -3,13 +3,67 @@
 
 package bsr
 
+import (
+	"testing"
+	"time"
+
+	"io"
+
+	"github.com/stretchr/testify/require"
+)
+
 func TestSessionRecordingMeta(s string, p Protocol) *SessionRecordingMeta {
 	return &SessionRecordingMeta{
 		Id:       s,
 		Protocol: p,
 	}
 }
 
+func TestSessionRecordingSummary(s string) *BaseSessionSummary {
+	return &BaseSessionSummary{
+		Id:              s,
+		ConnectionCount: 1,
+		StartTime:       time.Now(),
+		EndTime:         time.Now(),
+	}
+}
+
+func TestConnectionRecordingSummary(s string) *BaseConnectionSummary {
+	return &BaseConnectionSummary{
+		Id:           s,
+		ChannelCount: 1,
+		StartTime:    time.Now(),
+		EndTime:      time.Now(),
+		BytesUp:      256,
+		BytesDown:    256,
+	}
+}
+
+func TestConnectionRecordingMeta(s string) *ConnectionRecordingMeta {
+	return &ConnectionRecordingMeta{
+		Id: s,
+		channels: map[string]bool{
+			"test": true,
+		},
+	}
+}
+
+func TestChannelRecordingMeta(s string) *ChannelRecordingMeta {
+	return &ChannelRecordingMeta{
+		Id:   s,
+		Type: "chan",
+	}
+}
+
+// TestWriter should be used for testing a RequestWriter & MessageWriter
+func TestWriter(t *testing.T, w io.Writer) {
+	n, err := w.Write([]byte("hello world"))
+	require.NoError(t, err)
+	require.Equal(t, len("hello world"), n)
+	closer := w.(io.Closer)
+	require.NoError(t, closer.Close())
+}
```

Comment:
I am not sure this is a useful test helper. This is probably sufficient for the tests you are adding here, since the contents of the files are not important, but any external package looking to use this test helper would not get something that is usable as a request writer or message writer.

Reply:
I can change the helper function so that the parameter includes a `[]byte` type that gets passed into the Write function. This can help make it more useful. I can also add a comment explaining that the bytes needs to be a bsr chunk type.

Reply:
I think I would prefer just removing the test helper until there is obvious need/use for a helper that can encapsulate some complex and common test setup.